pandas/core/dtypes/dtypes.py

```diff
@@ -710,7 +710,8 @@ def __eq__(self, other):
             # None should match any subtype
             return True
         else:
-            return self.subtype == other.subtype
+            from pandas.core.dtypes.common import is_dtype_equal
+            return is_dtype_equal(self.subtype, other.subtype)
```

Comment:
This change was necessary since some subtypes don't nicely compare:
```python
In [2]: dtype1 = IntervalDtype('float64')

In [3]: dtype2 = IntervalDtype('datetime64[ns, US/Eastern]')

In [4]: dtype1 == dtype2
---------------------------------------------------------------------------
TypeError: data type not understood
```
Though interestingly enough the reverse comparison was fine:
```python
In [5]: dtype2 == dtype1
Out[5]: False
```

Reply:
yes this is because numpy types don't handle comparisons vs pandas dtypes.